pkg/api/dashboard.go

```diff
@@ -306,6 +308,26 @@ func (hs *HTTPServer) PostDashboard(c *models.ReqContext) response.Response {
 	if err := web.Bind(c.Req, &cmd); err != nil {
 		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
+
+	// Ideally, coremodel validation calls would be integrated into the web
+	// framework. But this does the job for now.
+	if cm, has := hs.CoremodelRegistry.Get("dashboard"); has {
```

Comment:
Yeah i don't like the strings either, as i [noted](https://github.com/grafana/grafana/pull/48252#discussion_r858572473). String lookup is useful for some cases - the comment mentions the k8sbridge case, which is an instance of the larger "a framework wants to use a coremodel to perform a generic act on a thing". But that's clearly not where we are when adding validation to an individual HTTP handler.

I do think we can make analyzable references _somehow_. My concerns have been:

* Limiting the number of places outside the coremodel component's subdirectory that need to be updated when changing/adding/removing a coremodel. Coremodels should be a strong contract, with a strong boundary.
* Ensuring that whatever approach works well with wire, and even beyond that, would be easy to rely on codegen to update whatever thing we have
* And/but/also need to be able to stub it out for feature flags, as this PR introduces

My current thoughts are inclined in the direction of adding a `type StructRegistry struct { ... }` to `pkg/framework/coremodel/staticregistry`, where there'd be a member of the struct for each coremodel, but have the type `coremodel.Interface` instead of coremodel-specific struct type. A typed constant like you suggest might be just as good, though. 

(Aside - your screenshot actually reminds me, i need to update the version of Thema here, too, because [i've begun improving error output](https://github.com/grafana/thema/pull/45))